exporter/exporterhelper/internal/persistent_queue.go

```diff
@@ -43,7 +42,10 @@ import (
 //	 index          index   x
 //	                        xxxx deleted
 type persistentQueue[T any] struct {
-	*queueCapacityLimiter[T]
+	// sizedElementsChannel is used by the persistent queue for two purposes:
+	// 1. a communication channel notifying the consumer that a new item is available.
+	// 2. capacity control based on the size of the items.
+	*sizedElementsChannel[permanentQueueEl]
```

Comment:
QQ: What is the memory footprint (size) of this channel? I can think of where we allow 1G elements on disk, would this be 1GB?

Reply:
That's a reasonable concern. I was thinking of addressing this by introducing the unbound channel, as I tried in the previous PR. But I can actually change the implementation a bit and bring back the `struct{}` channel for the persistent queue. Let me try that.

Reply:
Applied. PTAL